libs/model/src/utils/updateTags.ts

```diff
@@ -0,0 +1,62 @@
+import { AppError } from '@hicommonwealth/core';
+import { DB } from '@hicommonwealth/model';
+import { Op, Transaction } from 'sequelize';
+
+export const updateTags = async (
+  tag_ids: number[],
+  models: DB,
+  id: number | string,
+  idType: 'user_id' | 'community_id',
+  transaction?: Transaction,
+) => {
+  // tag_ids should be defined, even if its an empty array, then we remove existing tags
+  if (!tag_ids) return;
+
+  // verify all tag ids exist
+  if (tag_ids.length > 0) {
+    const tagCount = await models.Tags.count({
+      where: {
+        id: { [Op.in]: tag_ids },
+      },
+    });
+
+    const allTagsFound = tagCount === tag_ids.length;
+
+    if (!allTagsFound) {
+      throw new AppError('Some tag ids are invalid');
+    }
+  }
+
+  // remove all existing tags
+  idType === 'user_id'
+    ? await models.ProfileTags.destroy({ where: { user_id: id } })
+    : await models.CommunityTags.destroy({ where: { community_id: id } });
```

Comment:
Should these be part of the transaction?

Reply:
yes, good catch!